src/producer/partitioners/defaultJava/murmur2.spec.js

```diff
@@ -1,6 +1,7 @@
 const murmur2 = require('./murmur2')
 
-describe('Producer > Partitioner > DefaultJava > murmur2', () => {
+// TODO: sanity check our test input & implementation
```

Comment:
I'm very confident in the test input, since it's the same as the Java one (I even stepped through our implementation side by side with the Java one while developing it).

I would recommend creating a separate `murmur2` module that uses the `Long` module and then running the same test case in both, stepping through them line by line with interactive debugging to see where they start to diverge.